From a43ce4d3974b0a17c4d1f2e5aa6181da9d6eac77 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Mon, 14 Nov 2022 11:44:09 +0100
Subject: [PATCH 6/7] glibc2.31: Fixed what appears to be a genuine declaration
 error

db_desc_t should have been used really but I went for the equivalent
quick fix to touch the sources less.

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 nptl_db/fetch-value.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/nptl_db/fetch-value.c b/nptl_db/fetch-value.c
index 128d736adb..3b98785a57 100644
--- a/nptl_db/fetch-value.c
+++ b/nptl_db/fetch-value.c
@@ -140,7 +140,7 @@ _td_fetch_value (td_thragent_t *ta,
 
 td_err_e
 _td_store_value (td_thragent_t *ta,
-		 uint32_t desc[2], int descriptor_name, psaddr_t idx,
+		 uint32_t desc[3], int descriptor_name, psaddr_t idx,
 		 psaddr_t address, psaddr_t widened_value)
 {
   ps_err_e err;
@@ -240,7 +240,7 @@ _td_fetch_value_local (td_thragent_t *ta,
 
 td_err_e
 _td_store_value_local (td_thragent_t *ta,
-		       uint32_t desc[2], int descriptor_name, psaddr_t idx,
+		       uint32_t desc[3], int descriptor_name, psaddr_t idx,
 		       void *address, psaddr_t widened_value)
 {
   td_err_e terr = _td_locate_field (ta, desc, descriptor_name, idx, &address);
-- 
2.34.1

